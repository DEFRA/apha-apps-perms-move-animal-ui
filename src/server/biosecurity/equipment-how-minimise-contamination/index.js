import { QuestionPageController } from '../../common/controller/question-page-controller/question-page-controller.js'
import { EquipmentHowMinimiseContaminationAnswer } from '../../common/model/answer/equipment-how-minimise-contamination/equipment-how-minimise-contamination.js'
import { QuestionPage } from '../../common/model/page/question-page-model.js'
import { peopleDisinfectionPage } from '../people-disinfection/index.js'

const customHeading = 'Shared equipment'

export class EquipmentHowMinimiseContaminationPage extends QuestionPage {
  urlPath = '/biosecurity/equipment-how-minimise-contamination'
  sectionKey = 'biosecurity'
  question =
    'How will you minimise the risk of spread of TB infection to the incoming cattle when using shared equipment?'

  questionKey = 'equipmentHowMinimiseContamination'
  Answer = EquipmentHowMinimiseContaminationAnswer

  get heading() {
    return customHeading
  }

  view = 'biosecurity/equipment-how-minimise-contamination/index'

  nextPage() {
    return peopleDisinfectionPage
  }
}

export const equipmentHowMinimiseContaminationPage =
  new EquipmentHowMinimiseContaminationPage()

/**
 * @satisfies {ServerRegisterPluginObject<void>}
 */
export const equipmentHowMinimiseContamination = new QuestionPageController(
  equipmentHowMinimiseContaminationPage
).plugin()

/**
 * @import { ServerRegisterPluginObject } from '@hapi/hapi'
 */
