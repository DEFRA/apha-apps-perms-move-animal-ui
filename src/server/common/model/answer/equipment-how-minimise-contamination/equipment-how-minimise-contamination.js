import { TextAreaAnswer } from '../text-area/text-area.js'

/** @import {TextAreaConfig} from '../text-area/text-area.js' */

/**
 * @typedef {{ equipmentHowMinimiseContamination: string }} EquipmentHowMinimiseContaminationPayload
 */

/**
 * @augments {TextAreaAnswer<EquipmentHowMinimiseContaminationPayload>}
 */
export class EquipmentHowMinimiseContaminationAnswer extends TextAreaAnswer {
  /** @type {TextAreaConfig} */
  static config = {
    payloadKey: 'equipmentHowMinimiseContamination',
    rows: 8,
    isPageHeading: false,
    validation: {
      maxLength: {
        value: 5000,
        message: 'Your answer must be no longer than 5000 characters'
      },
      empty: {
        message:
          'Enter how you will minimise the risk of spread of TB infection to the incoming cattle when using shared equipment'
      }
    }
  }
}
