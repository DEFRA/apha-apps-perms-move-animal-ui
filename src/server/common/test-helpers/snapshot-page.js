import { parseDocument } from '../../common/test-helpers/dom.js'
import { createServer } from '~/src/server/index.js'
import { withCsrfProtection } from './csrf.js'
/** @import { Server } from '@hapi/hapi' */

/* global expect, it, beforeAll, afterAll, describe */

/**
 * @param {string} describeMessage
 * @param {string} shouldMessage
 * @param {string} urlPath
 */
export const describePageSnapshot = (
  describeMessage,
  shouldMessage,
  urlPath
) => {
  describe(describeMessage, () => {
    /** @type {Server} */
    let server

    beforeAll(async () => {
      server = await createServer()
      await server.initialize()
    })

    afterAll(async () => {
      await server.stop({ timeout: 0 })
    })

    it(shouldMessage, async () => {
      const { payload } = await server.inject(
        withCsrfProtection({
          method: 'GET',
          url: urlPath
        })
      )

      const content =
        parseDocument(payload).querySelector('#main-content')?.innerHTML

      expect(content).toMatchSnapshot()
    })
  })
}
