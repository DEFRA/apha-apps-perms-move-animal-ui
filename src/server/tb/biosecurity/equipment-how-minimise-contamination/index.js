import { TbQuestionPageController } from '../../question-page-controller.js'
import { EquipmentHowMinimiseContaminationAnswer } from '../../../common/model/answer/equipment-how-minimise-contamination/equipment-how-minimise-contamination.js'
import { QuestionPage } from '../../../common/model/page/question-page-model.js'
import { otherEquipmentMeasuresPage } from '../other-equipment-measures/index.js'
import { peopleDisinfectionPage } from '../people-disinfection/index.js'

export class EquipmentHowMinimiseContaminationPage extends QuestionPage {
  urlPath = '/biosecurity/equipment-how-minimise-contamination'
  sectionKey = 'biosecurity'
  question =
    'Which measures are in place to clean and disinfect equipment to reduce the risk of spreading TB?'

  questionKey = 'equipmentHowMinimiseContamination'
  Answer = EquipmentHowMinimiseContaminationAnswer

  /** @param {EquipmentHowMinimiseContaminationAnswer} answer */
  nextPage(answer) {
    if (answer.value?.includes('other')) {
      return otherEquipmentMeasuresPage
    } else {
      return peopleDisinfectionPage
    }
  }
}

export const equipmentHowMinimiseContaminationPage =
  new EquipmentHowMinimiseContaminationPage()

/**
 * @satisfies {ServerRegisterPluginObject<void>}
 */
export const equipmentHowMinimiseContamination = new TbQuestionPageController(
  equipmentHowMinimiseContaminationPage
).plugin()

/**
 * @import { ServerRegisterPluginObject } from '@hapi/hapi'
 */
