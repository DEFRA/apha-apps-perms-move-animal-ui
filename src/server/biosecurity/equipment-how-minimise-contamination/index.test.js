import { EquipmentHowMinimiseContaminationAnswer } from '../../common/model/answer/equipment-how-minimise-contamination/equipment-how-minimise-contamination.js'
import { describePageSnapshot } from '../../common/test-helpers/snapshot-page.js'
import { peopleDisinfectionPage } from '../people-disinfection/index.js'
import {
  equipmentHowMinimiseContaminationPage,
  EquipmentHowMinimiseContaminationPage
} from './index.js'

const sectionKey = 'biosecurity'
const question =
  'How will you minimise the risk of spread of TB infection to the incoming cattle when using shared equipment?'
const questionKey = 'equipmentHowMinimiseContamination'
const view = 'biosecurity/equipment-how-minimise-contamination/index'
const pageUrl = '/biosecurity/equipment-how-minimise-contamination'
const customHeading = 'Shared equipment'

describe('EquipmentHowMinimiseContaminationPage', () => {
  const page = new EquipmentHowMinimiseContaminationPage()

  it('should have the correct urlPath', () => {
    expect(page.urlPath).toBe(pageUrl)
  })

  it('should have the correct sectionKey', () => {
    expect(page.sectionKey).toBe(sectionKey)
  })

  it('should have the correct question', () => {
    expect(page.question).toBe(question)
  })

  it('should have the correct questionKey', () => {
    expect(page.questionKey).toBe(questionKey)
  })

  it('should have the correct heading', () => {
    expect(page.heading).toBe(customHeading)
  })

  it('should have the correct title', () => {
    expect(page.title).toBe(question)
  })

  it('should have the correct view', () => {
    expect(page.view).toBe(view)
  })

  it('should have the correct Answer model', () => {
    expect(page.Answer).toBe(EquipmentHowMinimiseContaminationAnswer)
  })

  it('nextPage should return any shared equipment page', () => {
    const nextPage = page.nextPage()
    expect(nextPage).toBe(peopleDisinfectionPage)
  })

  it('should export page', () => {
    expect(equipmentHowMinimiseContaminationPage).toBeInstanceOf(
      EquipmentHowMinimiseContaminationPage
    )
  })

  describePageSnapshot({
    describes: 'equipmentHowMinimiseContaminationPage.content',
    it: 'should render expected response and content',
    pageUrl
  })
})
