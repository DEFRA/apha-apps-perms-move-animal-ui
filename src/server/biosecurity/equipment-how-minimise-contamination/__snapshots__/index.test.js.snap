// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EquipmentHowMinimiseContaminationPage equipmentHowMinimiseContaminationPage.content should render expected response and content 1`] = `
"
  <div class="govuk-grid-row">
    <div class="govuk-grid-column-two-thirds">
      <h1 class="govuk-heading-l">Shared equipment</h1>
    </div>
  </div>
  <div class="govuk-grid-row">
    <div class="govuk-grid-column-two-thirds">
      <p>
        To prevent the spread of TB, cleaning and disinfection measures for
        shared equipment should be in place.
      </p>
      <p>This can include:</p>
      <ul class="govuk-list govuk-list--bullet">
        <li>designated disinfection points</li>
        <li>cleaning and disinfecting shared machinery</li>
        <li>
          not sharing equipment or machinery between the groups of incoming and
          resident cattle
        </li>
        <li>
          cleaning and disinfection measures for shared milking and handling
          facilities
        </li>
      </ul>
    </div>
  </div>
    <form method="POST" novalidate="">
      <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
          <input type="hidden" name="crumb" value="csrf-value">
          <input type="hidden" name="nextPage" value="">

<fieldset class="govuk-fieldset">
  <div class="govuk-form-group">
  <label class="govuk-label govuk-label--m" for="equipmentHowMinimiseContamination">
    How will you minimise the risk of spread of TB infection to the incoming cattle when using shared equipment?
  </label>
  <textarea class="govuk-textarea" id="equipmentHowMinimiseContamination" name="equipmentHowMinimiseContamination" rows="8"></textarea>
</div>


</fieldset>
            <button type="submit" class="govuk-button" data-module="govuk-button" id="continue-button">
  Continue
</button>
        </div>
      </div>
    </form>
        "
`;
